src/gpu/generic/sycl/ref_matmul.hpp

```diff
@@ -0,0 +1,158 @@
+            const memory_desc_wrapper bias_d(weights_md(1));
+            const memory_desc_wrapper dst_d(dst_md());
+
+            const bool ok = set_default_params() == status::success
+                    && attr_.set_default_formats(dst_md()) == status::success
+                    && check_data_types(src_d, weights_d, dst_d)
+                    && check_formats(src_d, weights_d, dst_d)
+                    && attr()->has_default_values(sm::scales_runtime
+                            | sm::zero_points_runtime | sm::post_ops
+                            | sm::dropout | sm::scales_runtime_data_type
+                            | sm::zero_points_runtime_data_type)
+                    && IMPLICATION(!attr()->scales_.has_default_values(),
+                            check_scales_mask())
+                    && post_ops_ok() && md_dims_in_range(src_md())
+                    && md_dims_in_range(weights_md());
+            if (!ok) return status::unimplemented;
+
+            return init_conf();
+        }
+
+        sycl_matmul_conf_t conf_;
+
+    private:
+        status_t init_conf();
+
+        status_t set_default_params() {
+            if (src_md_.format_kind == format_kind::any) {
+                auto src_tag = utils::pick(ndims() - 2, format_tag::ab,
+                        format_tag::abc, format_tag::abcd);
+                CHECK(memory_desc_init_by_tag(src_md_, src_tag));
+            }
+            const memory_desc_wrapper src_d(src_md());
+            if (src_d.is_blocking_desc()) {
+                if (weights_md_.format_kind == format_kind::any) {
+                    CHECK(memory_desc_init_by_blocking_desc(
+                            weights_md_, src_d.blocking_desc()));
+                }
+                if (dst_md_.format_kind == format_kind::any) {
+                    CHECK(memory_desc_init_by_blocking_desc(
+                            dst_md_, src_d.blocking_desc()));
+                }
+            }
+            const memory_desc_wrapper dst_d(dst_md());
+            if (dst_d.is_blocking_desc()) {
+                if (bias_md_.format_kind == format_kind::any) {
+                    CHECK(memory_desc_init_by_blocking_desc(
+                            bias_md_, dst_d.blocking_desc()));
+                }
+            }
+            return status::success;
+        }
+
+        bool check_scales_mask() const {
+            const std::vector<int> supported_args
+                    = {DNNL_ARG_SRC_0, DNNL_ARG_WEIGHTS_0, DNNL_ARG_DST};
+            return attr_scales_ok(supported_args);
+        }
+
+        bool post_ops_ok() const {
+            // Dw conv post-ops are not supported.
+            return attr()->post_ops_.len() <= sycl_post_ops_t::max_post_ops
+                    && attr()->post_ops_.has_default_values(
+                            {primitive_kind::eltwise, primitive_kind::binary,
+                                    primitive_kind::sum});
+        }
```

Comment:
Does it makes sense to rebase on top of #2058 , so that post_ops check relies on sycl_post_ops::post_ops_ok from the get-go?

Reply:
No need to merge both PRs. Just wanted to highlight that whichever gets merged first, the second will need to be amended accordingly.

Reply:
Sorry, what I wrote was ambiguous. I meant if you want to merge both PRs at the same time, it makes sense that I do that. Otherwise I need to ament the second one after you merge the first one.

Reply:
@mgouicem do you intend to merge both PRs at roughly the same time? Should I rebase one on top of another?

Reply:
@dzarukin  will be handling it. 
In general, it is simpler to process multiple PRs at once as it reduces overheads of synchronizing internal and public repositories.